libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/backpress/BackPressedComponent.kt

```diff
@@ -0,0 +1,84 @@
+package com.mapbox.navigation.dropin.backpress
+
+import androidx.activity.ComponentActivity
+import androidx.activity.OnBackPressedCallback
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.repeatOnLifecycle
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.NavigationView
+import com.mapbox.navigation.ui.app.internal.Store
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.endNavigation
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.base.lifecycle.UIComponent
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onCompletion
+import kotlinx.coroutines.launch
+
+/**
+ * Back pressed callback for the [NavigationView].
+ *
+ * On back pressed will update the navigation state.
+ *
+ * (FreeDrive) <- (DestinationPreview) <- (RoutePreview) <- (ActiveNavigation)
+ *             <- (Arrival)
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+internal class BackPressedComponent(
+    componentActivity: ComponentActivity,
```

Comment:
I don't think we need to inject the ComponentActivity here. All we need is `OnBackPressedDispatcher`.